hxmath/math/DualQuaternion.hx

```diff
@@ -0,0 +1,622 @@
+     * @param axis          The axis to rotate around.
+     * @param t             The Vector4 ( translation )
+     * @return              The DualQuaternion
+     */
+    public static inline function fromAxisRadian( theta:Float
+                                                , axis: Vector3
+                                                , vec4: Vector4 ):DualQuaternion 
+    {
+        var cosHalfAngle = Math.cos(theta / 2.0);
+        var sinHalfAngle = Math.sin(theta / 2.0);
+        var real = new Quaternion(cosHalfAngle,sinHalfAngle * axis.x,sinHalfAngle * axis.y,
+        sinHalfAngle * axis.z);
+        return create(real,vec4);
+    }
+    
+    /**
+     * Extract the Vector4 part of a DualQuaternion
+     * 
+     * @return          The translation Vector
+     */
+    public inline function getTranslation(): Vector4 
+    {
+        var q = this.dual * (~this.real) * 2.0;
+        return new Vector4(q.x, q.y, q.z, 1.0);
+    }
+    
+    /**
+     * Extract the Real Quaternion the 3D angle as a copy
+     * Particularly useful as it clarifies that the Rotation
+     * 
+     * @return         The rotational Quaternion
+     */
+    public inline function getRotation(): Quaternion 
+    {
+        return this.real.clone();
+    }
+    
+    /**
+     * Construct a DualQuaternion from an array.
+     * 
+     * @param rawData   The input array.
+     * @return          The constructed structure.
+     */
+    @:from
+    public static inline function fromArray(rawData:Array<Float>):DualQuaternion
+    {
+        if (rawData.length != DualQuaternion.elementCount)
+        {
+            throw "Invalid rawData.";
+        }
+        var real = new Quaternion(rawData[0], rawData[1], rawData[2], rawData[3]);
+        var dual = new Quaternion(rawData[4], rawData[5], rawData[6], rawData[7]);
+        return new DualQuaternion(real, dual);
+    }
+    
+    /**
+     * Creates a Scalar DualQuaternion
+     * 
+     * @param s         The scale value
+     * @return          The Scalar DualQuaternion
+     */
+    public static inline function scalar(s: Float):DualQuaternion
+    {
+        return new DualQuaternion(Quaternion.scalarReal( s ), Quaternion.zero);
+    }
+    
+    /**
+     * Clone.
+     * 
+     * @return          The cloned object.
+     */
+    public inline function clone():DualQuaternion
+    {
+        return new DualQuaternion( this.real.clone(), this.dual.clone() );
+    }
+    
+    /**
+     * dot product
+     *
+     * @param a         one DualQuaternion
+     * @param b         another DualQuaternion
+     * @return          the dot product.
+     */
+    public static inline function dot(a:DualQuaternion, b:DualQuaternion):Float
+    {
+        return Quaternion.dot(a.real.clone(), b.real.clone());
+    }
+    
+    /**
+     * Create a complex conjugate copy of a dualquaternion
+     * 
+     * @param a
+     * @return       ~a
+     */
+    @:op(~A)
```

Comment:
Like I said in my previous comment, this might not make the most sense for choice of conjugate. Here's my reasoning:

* Rotation by a unit length quaternion q is calculated as q * v * (~q) (where ~q is the complex conjugate).
* With this convention, transformation by a dual quaternion q would be written q * v * (~q.dualConjugate()) which is kind of gross (also, dualConjugate doesn't exist yet, so you can't actually write this).